src/main/java/com/google/cloud/spanner/jdbc/JdbcParameterMetaData.java

```diff
@@ -29,9 +35,23 @@
 class JdbcParameterMetaData extends AbstractJdbcWrapper implements ParameterMetaData {
   private final JdbcPreparedStatement statement;
 
-  JdbcParameterMetaData(JdbcPreparedStatement statement) throws SQLException {
+  private final StructType parameters;
+
+  JdbcParameterMetaData(JdbcPreparedStatement statement, ResultSet resultSet) {
     this.statement = statement;
-    statement.getParameters().fetchMetaData(statement.getConnection());
+    this.parameters = resultSet.getMetadata().getUndeclaredParameters();
```

Comment:
In this case the name of this field in the proto definition is a bit unfortunate. It says 'undeclared parameters', but it actually returns information about all parameters, including the ones that included a type and/or value when it was sent by the client. So if the user included type information for one or more parameters, it will also show up in this map.